index.html

```diff
@@ -4,574 +4,543 @@
-            Let |chunk| be a newly created {{Uint8Array}} over the slice of
-            |buffer| from `0` to |bytesRead|.
-          <li>
-            Invoke |controller|.
-            <a data-cite="streams#rs-default-controller-enqueue">`enqueue()`</a>
-            with |chunk|.
-          </li>
-        </ol>
-      <li>
-        If a buffer overrun condition was encountered, invoke
-        |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with a
-        <dfn>`BufferOverrunError`</dfn>.
-      </li>
-      <li>
-        If a break condition was encountered, invoke |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with a
-        <dfn>`BreakError`</dfn>.
-      </li>
-      <li>
-        If a framing error was encountered, invoke |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with a
-        <dfn>`FramingError`</dfn>.
-      </li>
-      <li>
-        If a parity error was encountered, invoke |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with a
-        <dfn>`ParityError`</dfn>.
-      </li>
-      <li>
-        If an operating system error was encountered, invoke
-        |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with an
-        {{UnknownError}}.
-      </li>
-      <li>
-        If |port| was disconnected, invoke |controller|.
-        <a data-cite="streams#rs-default-controller-error">`error()`</a> with a
-        {{NetworkError}}.
-      </li>
-    </ol>
-  </li>
-</ol>
-
-<p class="note">
-The {{Promise}} returned by this algorithm is immediately resolved so that it
-does not block canceling the stream. [[STREAMS]] specifies that this algorithm
-will not be invoked again until a chunk is enqueued.
-</p>
-
-## <dfn>SerialPortSourceCancelAlgorithm</dfn>(|port:SerialPort|)
-
-1. Let |promise| be [=a new promise=].
-1. [=In parallel=], invoke the operating system to discard the contents of all
-   software and hardware receive buffers for |port| and [=resolve=] |promise|.
-1. Return |promise|.
+  ## Serial port source abstract operations
+
+  ## <dfn>SerialPortSourcePullAlgorithm</dfn>(|port:SerialPort|, |controller:ReadableStreamDefaultController|)
+
+  1. Let |promise:Promise| be [=a new promise=].
+  1. [=Resolve=] |promise|.
+  1. Return |promise| and run the remaining steps [=in parallel=]:
+    1. Let |desiredSize| be |controller|.{{`desiredSize`}}
+    1. Let |buffer| be a newly created {{ArrayBuffer}} of |desiredSize| bytes.
+    1. Invoke the operating system to read up to |desiredSize| bytes from |port| into |buffer|.
+    1. If no errors were encountered run the following steps:
+      1. Let |bytesRead| be the number of bytes written to |buffer|.
+      1. Let |chunk| be a newly created {{Uint8Array}} over the slice of
+          |buffer| from `0` to |bytesRead|.
+      1. Invoke |controller|.{{`enqueue()`}} with |chunk|.
```

Comment:
Use `{{ReadableStreamDefaultController/enqueue()}}` to get proper linking. Same for `error()` below.

Reply:
Fixed in https://github.com/reillyeon/serial/pull/2/commits/46408a109e4dfd5afc5ca50b0d483de7ec74ec94